# Demographic Data Analyzer

## Import the data and extract top 5 data

In [2]:
import pandas as pd
dat = pd.read_csv('adult.data.csv')
dat.head()


age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [3]:
dat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  salary          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


## How many people of each race are represented in this dataset?

In [4]:
dat['race'].value_counts()

race
White                 27816
Black                  3124
Asian-Pac-Islander     1039
Amer-Indian-Eskimo      311
Other                   271
Name: count, dtype: int64

The number of people who join in White race is 27.816, Black race is 3124, Asian-Pac-Islander is 1039, Amer-Indian-Eskimo is 311, and 271 people join other race.

## What is the average age of men?

In [6]:
round(dat[dat['sex'] == 'Male']['age'].mean(),0)

39.0

The average age of men in this dataset is 39 years old

## What is the percentage of people who have a Bachelor's degree?

In [7]:
# Peple who have a bachelor's degree
len(dat[dat['education'] == "Bachelors"])
# Percentage
print(round((len(dat[dat['education'] == "Bachelors"])/len(dat))*100,2),"%")

16.45 %


So, the percentage of people who have a bachelor's degree is 16.45% or we can say nearly 16 people from 100 people have a bachelor's degree

## What percentage of people with advanced education (Bachelors, Masters, or Doctorate) make more than 50K?

In [8]:
print(round(len(dat[dat['education'].isin(['Bachelors','Masters','Doctorate']) & dat['salary'].isin(['>50K'])])/len(dat)*100,2),"%")

10.71 %


So, only 10.7% people with advanced education who can make more than 50K salary.

## What is the minimum number of hours a person works per week?

In [9]:
min_hour = dat['hours-per-week'].min()
min_hour

1

minimum number of hours a person workks per week is 1 hour.

## What percentage of the people who work the minimum number of hours per week have a salary of more than 50K?

In [10]:
print(round(len(dat[(dat['hours-per-week'] == min_hour) & (dat['salary'] == '>50K')])/len(dat)*100,2),"%")

0.01 %


Only 0.01% people who have 1 hour per week and get more than $50K of salary. 

## What country has the highest percentage of people that earn >50K and what is that percentage?

In [11]:
dat.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [12]:
dat_co = dat[dat['salary'] == '>50K']['native-country'].value_counts().reset_index()
dat_co['perc'] = round(dat_co['count']/len(dat)*100,2)
dat_co.head()

native-country  count   perc
0  United-States   7171  22.02
1              ?    146   0.45
2    Philippines     61   0.19
3        Germany     44   0.14
4          India     40   0.12

In [13]:
print("United-States is the highest percentage of people that earn > 50K with 22.02 %")

United-States is the highest percentage of people that earn > 50K with 22.02 %


## Identify the most popular occupation for those who earn >50K in India.

In [14]:
dat.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [15]:
dat[(dat['native-country'] == 'India') & (dat['salary'] == '>50K')]['occupation'].value_counts()

occupation
Prof-specialty      25
Exec-managerial      8
Other-service        2
Tech-support         2
Transport-moving     1
Sales                1
Adm-clerical         1
Name: count, dtype: int64

The popular occupation who earn > 50 K in India is Prof-specialty